spec/services/iiif_print/tenant_config_spec.rb

```diff
@@ -0,0 +1,188 @@
+        end
+      end
+    end
+
+    describe '#create_derivatives' do
+      subject { instance.create_derivatives("filename") }
+
+      context 'when the feature is flipped to false' do
+        before { test_strategy.switch!(:use_iiif_print, false) }
+
+        it 'raises an error' do
+          expect { subject }.to raise_error(IiifPrint::TenantConfig::LeakyAbstractionError)
+        end
+      end
+
+      context 'when the feature is flipped to true' do
+        before { test_strategy.switch!(:use_iiif_print, true) }
+        it 'delegates to the configured iiif_service' do
+          expect(instance.iiif_print_service_instance).to receive(:create_derivatives)
+          subject
+        end
+      end
+    end
+
+    describe '#cleanup_derivatives' do
+      subject { instance.cleanup_derivatives }
+
+      context 'when the feature is flipped to false' do
+        before { test_strategy.switch!(:use_iiif_print, false) }
+
+        it 'raises an error' do
+          expect { subject }.to raise_error(IiifPrint::TenantConfig::LeakyAbstractionError)
+        end
+      end
+
+      context 'when the feature is flipped to true' do
+        before { test_strategy.switch!(:use_iiif_print, true) }
+        it 'delegates to the configured iiif_service' do
+          expect(instance.iiif_print_service_instance).to receive(:cleanup_derivatives)
+          subject
+        end
+      end
+    end
+  end
+
+  describe IiifPrint::TenantConfig::PdfSplitter do
+    describe '.iiif_print_splitter' do
+      subject { described_class.iiif_print_splitter }
+
+      it { is_expected.to eq(::IiifPrint::SplitPdfs::PagesToJpgsSplitter) }
+    end
+
+    describe '.call' do
+      subject { described_class.call(:arg) }
+
+      context 'when the feature is flipped to false' do
+        before { test_strategy.switch!(:use_iiif_print, false) }
+
+        it { is_expected.to eq([]) }
+      end
+
+      context 'when the feature is flipped to true' do
+        before { test_strategy.switch!(:use_iiif_print, true) }
+
+        it 'delegates to the configured .iiif_print_splitter' do
+          expect(described_class.iiif_print_splitter).to receive(:call).with(:arg)
+          subject
+        end
+      end
+    end
+  end
+
+  describe IiifPrint::TenantConfig::SkipSplittingPdfService do
+    describe '.conditionally_enqueue' do
+      subject { described_class.conditionally_enqueue }
+
+      it { is_expected.to eq(:tenant_does_not_split_pdfs) }
+    end
+  end
+
+  describe Hyrax::Actors::FileSetActor do
+    let(:instance) { described_class.new(:file_set, :user) }
+
+    ##
+    # The purpose of this spec is to demonstrate that load sequence of modules correctly evaluates
+    describe '#service' do
+      subject { instance.service }
+
+      context 'when the feature is flipped to false' do
+        before { test_strategy.switch!(:use_iiif_print, false) }
+
+        it { is_expected.to eq(IiifPrint::SplitPdfs::ChildWorkCreationFromPdfService) }
```

Comment:
Doesn't false on the flipper mean you do NOT use IiifPrint, in which case the service would be the TenantConfig service?